src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -307,6 +388,7 @@ private String buildOAuthRedirectUrl() {
      * This is where the user comes back to at the end of the OpenID redirect ping-pong.
      */
     public HttpResponse doFinishLogin(StaplerRequest request) throws IOException {
+    	LOGGER.fine("Realm do finish " + request);
```

Comment:
> Are logging messages supposed to prefix by the method that is calling them? It's inconsistently applied in your debugging.

Best practice is usually:
```java
if logger.isloggable(level) logger.logp(level, class, method, format, args)
```
That does a cheap check before building args for the logging method.  And the logging method doesn't have to perform an expensive stack inspection to determine the caller class/function.

If all you want is a, "did I call" point, you can use `logger.entering`.